### PR TITLE
ELMU #135 Add password confirmation

### LINKED ISSUE #135: Confirm account data update with password
On the account page, changing the username and/or email data from the 'Account' section should prompt the user to type in their password and re-confirm their identity.

### PR DESCRIPTION
**Pull request by laurentiu-ilici**
Resolves #135

### CODE REVIEW

File: src/components/confirmation-dialogs.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import React from 'react';
-import { Modal, Input, Checkbox } from 'antd';
+import { Modal, Input, Checkbox, Form } from 'antd';
 
 const confirm = Modal.confirm;
 const TextArea = Input.TextArea;
+const FormItem = Form.Item;
 
-export function confirmSectionDelete(t, section, onOk, onCancel = () => {}) {
+export function confirmSectionDelete(t, section, onOk, onCancel = () => { }) {
```
Comment (ahelmberger) [inline src/components/confirmation-dialogs.js:8]: I do not really care, I'm just wondering what made you add this space here (and in the other places ?

File: src/components/confirmation-dialogs.js [hunk 2]
```diff
@@ -104,3 +105,64 @@ export function confirmDocumentRevisionRestoration(t, revision, onOk, onCancel =
...
+  let errorMessage = '';
+
+  const handleConfirmPassword = async password => {
+    try {
+      const { user } = await userApiClient.login({ username, password });
+      if (user) {
+        validationStatus = 'success';
+        errorMessage = '';
+        onOk();
+        return Promise.resolve();
```
Comment (ahelmberger) [inline src/components/confirmation-dialogs.js:123]: just `return;`, async functions already return a promise, no need to create an additional one.

File: src/components/confirmation-dialogs.js [hunk 3]
```diff
@@ -104,3 +105,64 @@ export function confirmDocumentRevisionRestoration(t, revision, onOk, onCancel =
...
+      }
+
+      validationStatus = 'error';
+      errorMessage = t('confirmationDialogs:wrongPasswordProvided');
+      dialog.update(createDialogProps());
+      return Promise.reject(new Error('Wrong password'));
```
Comment (ahelmberger) [inline src/components/confirmation-dialogs.js:129]: here just `throw new Error(...)` (but I'm not sure why we have to throw at all?)

File: src/components/confirmation-dialogs.js [hunk 4]
```diff
@@ -104,3 +105,64 @@ export function confirmDocumentRevisionRestoration(t, revision, onOk, onCancel =
...
+    } catch (error) {
+      Modal.error({ title: t('common:error'), content: error.message });
+      return Promise.resolve();
```
Comment (ahelmberger) [inline src/components/confirmation-dialogs.js:132]: This line can be deleted

File: src/components/confirmation-dialogs.yml [hunk 5]
```diff
@@ -16,3 +16,15 @@ deleteAllDescendantRevisions:
...
+  de: Kennwortbestätigung
+enterPassword:
+  en: Please enter a password
+  de: Bitte geben Sie ein Kennwort an
+password:
+  en: Passsword
+  de: Kennwort
+wrongPasswordProvided:
+  en: The wrong password was provided
+  de: Falsches Passwort
```
Comment (ahelmberger) [inline src/components/confirmation-dialogs.yml:30]: "Kennwort", we never use "Passwort" I think. (Maybe even better: "Das Kennwort ist ungültig")

File: src/components/confirmation-dialogs.yml [hunk 6]
```diff
@@ -16,3 +16,15 @@ deleteAllDescendantRevisions:
 restoreDocumentRevisionConfirmation:
   en: Do you really want to restore the content of revision {revisionId}?
   de: Möchten Sie den Inhalt der Revision {revisionId} wirklich wiederherstellen?
+confirmPassword:
+  en: Confirm password
+  de: Kennwortbestätigung
```
Comment (adrianaluchian) [inline src/components/confirmation-dialogs.yml:21]: Small thing, should it be 'Kennwort bestätigen' or 'Password confirmation', to match the phrasings between languages? 🤔

Comment (ahelmberger) [reply src/components/confirmation-dialogs.yml:21]: I think it's a label for an input, so I would tend to use "Password confirmation" and "Kennwortbestätigung"

File: src/components/confirmation-dialogs.js (hunk 1 above)
Comment (laurentiu-ilici) [reply src/components/confirmation-dialogs.js:8]: The code editor does that automagically, I think there's something left in the settings, I'll try to find it.

File: src/components/confirmation-dialogs.js (hunk 3 above)
Comment (laurentiu-ilici) [reply src/components/confirmation-dialogs.js:129]: We need to reject the promise.

File: src/components/confirmation-dialogs.js (hunk 1 above)
Comment (ahelmberger) [reply src/components/confirmation-dialogs.js:8]: It's not a biggie, we just never have that space anywhere else.

File: src/components/confirmation-dialogs.js (hunk 3 above)
Comment (ahelmberger) [reply src/components/confirmation-dialogs.js:129]: True. Then we should throw without catching it (either outside the try or rethrow in the catch block). I'd rather not create promises manually inside async functions, it's a little bit confusing to see why.

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**